### PR TITLE
Change Source Architecture and Improve Performance

### PR DESCRIPTION
**Pull request by niruhan**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: component/src/main/java/io/siddhi/extension/io/grpc/sink/GrpcSink.java [hunk 1]
```diff
@@ -202,87 +212,174 @@
...
-            @Override //todo latch based error???
-            public void onError(Throwable t) { //parent method doest have error in its signature. so cant throw
-                // from here
+                @Override
+                public void onError(Throwable t) { //parent method doest have error in its signature. so cant throw
+                    // from here
 //                    if (((StatusRuntimeException) t).getStatus().getCode().equals(Status.UNAVAILABLE)) {
 //                        throw new ConnectionUnavailableException(siddhiAppName.getName() + ": " + streamID + ": "
 //                        + t.getMessage());
 //                    }
```
Comment (mohanvive) [inline component/src/main/java/io/siddhi/extension/io/grpc/sink/GrpcSink.java]: Remove commented code.

File: component/src/test/java/io/siddhi/extension/io/grpc/proto/StreamServiceGrpc.java [hunk 2]
```diff
@@ -0,0 +1,314 @@
+package io.siddhi.extension.io.grpc.proto;
```
Comment (mohanvive) [inline component/src/test/java/io/siddhi/extension/io/grpc/proto/StreamServiceGrpc.java]: Add license header.

File: docs/api/1.0.0-SNAPSHOT.md [hunk 3]
```diff
@@ -0,0 +1,981 @@
+# API Docs - v1.0.0-SNAPSHOT
```
Comment (mohanvive) [inline docs/api/1.0.0-SNAPSHOT.md]: Remove this file

File: pom.xml [hunk 4]
```diff
@@ -189,36 +201,68 @@
...
+        <siddhi.map.protobuf.version>1.0.0-SNAPSHOT</siddhi.map.protobuf.version>
         <jacoco.plugin.version>0.7.9</jacoco.plugin.version>
     </properties>
-
-    <scm>
-        <url>https://github.com/siddhi-io/siddhi-io-grpc.git</url>
-        <developerConnection>scm:git:https://github.com/siddhi-io/siddhi-io-grpc.git</developerConnection>
-        <connection>scm:git:https://github.com/siddhi-io/siddhi-io-grpc.git</connection>
-        <tag>HEAD</tag>
-    </scm>
```
Comment (mohanvive) [inline pom.xml]: Don't remove scm tags.

File: component/src/main/java/io/siddhi/extension/io/grpc/source/GrpcEventServiceServer.java [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+                        "creating SslContext. " + e.getMessage(), e);
+            }
+        }
+        if (grpcServerConfigs.getMaxInboundMessageSize() != -1) {
+            serverBuilder.maxInboundMessageSize(grpcServerConfigs.getMaxInboundMessageSize());
+        }
+        if (grpcServerConfigs.getMaxInboundMetadataSize() != -1) {
+            serverBuilder.maxInboundMetadataSize(grpcServerConfigs.getMaxInboundMetadataSize());
+        }
+
```
Comment (mohanvive) [inline component/src/main/java/io/siddhi/extension/io/grpc/source/GrpcEventServiceServer.java]: Remove line space.

File: component/src/main/java/io/siddhi/extension/io/grpc/sink/AbstractGrpcSink.java [hunk 6]
```diff
@@ -69,25 +62,19 @@
...
+    protected Map<String, String> headersMap;
 
     /**
      * Returns the list of classes which this sink can consume.
      * Based on the type of the sink, it may be limited to being able to publish specific type of classes.
      * For example, a sink of type file can only write objects of type String .
-     *
      * @return array of supported classes , if extension can support of any types of classes
      * then return empty array .
      */
```
Comment (mohanvive) [inline component/src/main/java/io/siddhi/extension/io/grpc/sink/AbstractGrpcSink.java]: Comments should be align with the IO context that you are implementing.

File: component/src/main/java/io/siddhi/extension/io/grpc/sink/GrpcSink.java [hunk 7]
```diff
@@ -202,87 +212,174 @@
...
+                public void onNext(Empty event) {
+                }
+
+                @Override
+                public void onError(Throwable t) { //parent method doest have error in its signature. so cant throw
+                    // from here
+//                    if (((StatusRuntimeException) t).getStatus().getCode().equals(Status.UNAVAILABLE)) {
+//                        throw new ConnectionUnavailableException(siddhiAppName.getName() + ": " + streamID + ": "
+//                        + t.getMessage());
+//                    }
```
Comment (mohanvive) [inline component/src/main/java/io/siddhi/extension/io/grpc/sink/GrpcSink.java]: Remove commented code block.

File: pom.xml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
-~ Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+~ Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (ramindu90) [inline pom.xml]: @niruhan this should be changed as 2019, is it?

Comment (niruhan) [reply pom.xml]: Fixed in https://github.com/siddhi-io/siddhi-io-grpc/pull/26

**Pull request merged**